ui/lib/sync/addon/components/secrets/page/destinations.ts

```diff
@@ -28,15 +30,31 @@ export default class SyncSecretsDestinationsPageComponent extends Component<Args
   @service declare readonly store: StoreService;
   @service declare readonly flashMessages: FlashMessageService;
 
+  // for some reason there isn't a full page refresh happening when transitioning on filter change
+  // when the transition happens it causes the FilterInput component to lose focus since it can only focus on didInsert
+  // to work around this, verify that a transition from this route was completed and then focus the input
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    this.router.on('routeDidChange', this.focusNameFilter);
+  }
+
+  willDestroy(): void {
+    super.willDestroy();
+    this.router.off('routeDidChange', this.focusNameFilter);
+  }
+
+  focusNameFilter(transition?: Transition) {
+    const route = 'vault.cluster.sync.secrets.destinations.index';
+    if (transition?.from?.name === route && transition?.to?.name === route) {
+      next(() => document.getElementById('name-filter')?.focus());
+    }
+  }
```

Comment:
I spent a good deal of time trying to find a more global way of dealing with this to no avail. It's very strange since pretty much the exact same workflow happens for LDAP roles and when the transition happens the page does a full refresh, causing a  re-render of the `FilterInput` which triggers a focus on `didInsert`.

In sync we get a partial re-render so the component isn't torn down. This was the best I could come up with that would be based on the actual condition. Simply doing `later(() => document.getElementById('name-filter')?.focus(), 100)` immediately after the transition in the `onFilterChange` method did the trick but felt like a flaky fix. This at least is triggered by the completion of the transition.